api/src/utils/media/csv/validation/csv-row-validator.ts

```diff
@@ -352,3 +352,55 @@ export const getValidFormatFieldsValidator = (config?: ColumnFormatValidatorConf
     return csvWorksheet;
   };
 };
+
+export type FileColumnUniqueValidatorConfig = {
+  file_column_unique_validator: {
+    key_columns: string[];
+  };
+};
+
+export const getUniqueColumnsValidator = (config?: FileColumnUniqueValidatorConfig): CSVValidator => {
+  return (csvWorksheet) => {
+    if (!config) {
+      return csvWorksheet;
+    }
+
+    if (config.file_column_unique_validator.key_columns.length < 1) {
+      return csvWorksheet;
+    }
+
+    const keySet = new Set();
+    const rows = csvWorksheet.getRowObjects();
+    const lowercaseHeaders = csvWorksheet.getHeadersLowerCase();
+
+    const columnIndices = config.file_column_unique_validator.key_columns.map((column) =>
+      lowercaseHeaders.indexOf(column.toLocaleLowerCase())
+    );
+    if (columnIndices.indexOf(-1) !== -1) {
+      // worksheet or config columns don't line up
+      return csvWorksheet;
+    }
+
+    rows.forEach((row, rowIndex) => {
+      const key = config.file_column_unique_validator.key_columns.map((columnIndex) => row[columnIndex]).join(', ');
```

Comment:
Might want to sanitize any white space using `.map(s => s.trim())` before serializing